### PR TITLE
storagenode: more live request tracking

### PR DESCRIPTION
**Pull request by jtolds**
What: show grpc errors in monkit data and add more tracking around storage node live requests

Why: when an upload is rejected we return the Unavailable status code over grpc, but monkit doesn't know how to show grpc status codes, so we don't know how many unavailable requests are getting returned. first, add grpc status codes to monkit so monkit knows and can show us. second - we can see stats about individual types of live requests, but add tracking for all of them

Please describe the tests: no functionality change, just additional instrumentation

Please describe the performance impact: very, very slight - a couple new monkit compare and swaps for upload/download and a couple of new function calls during request errors

## Code Review Checklist (to be filled out by reviewer)
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: pkg/process/error.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+package process
+
+import (
+	"google.golang.org/grpc/status"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+)
+
+func init() {
+	monkit.AddErrorNameHandler(func(err error) (string, bool) {
```
Comment (stefanbenten) [inline pkg/process/error.go]: Not entirely sure what this error handler adds as a benefit.
Would you explain that for me?

Comment (stefanbenten) [reply pkg/process/error.go]: Do you want to add additional metrics what grpc error is being returned?

Comment (jtolds) [reply pkg/process/error.go]: i tried to explain this in the PR description - this part:
> when an upload is rejected we return the Unavailable status code over grpc, but monkit doesn't know how to show grpc status codes, so we don't know how many unavailable requests are getting returned. first, add grpc status codes to monkit so monkit knows and can show us.

Comment (stefanbenten) [reply pkg/process/error.go]: Is that function not run for every err that is being returned by any function?
Not sure about the overhead there.

Comment (jtolds) [reply pkg/process/error.go]: you're absolutely right, but errors (hopefully! ha) don't happen that often.

**Pull request merged**